code/go/0chain.net/smartcontract/dbs/types.go

```diff
@@ -7,30 +7,39 @@ import (
 )
 
 type DbHealthCheck struct {
-	ID				string			 `json:"id"`
+	ID              string           `json:"id"`
 	LastHealthCheck common.Timestamp `json:"last_health_check"`
-	Downtime		uint64			 `json:"downtime"`
+	Downtime        uint64           `json:"downtime"`
 }
 
 type DbUpdates struct {
 	Id      string                 `json:"id"`
 	Updates map[string]interface{} `json:"updates"`
 }
 
-func NewDbUpdates(id string) *DbUpdates {
-	return &DbUpdates{
-		Id:      id,
-		Updates: make(map[string]interface{}),
+type DbUpdateProvider struct {
+	DbUpdates
+	Type spenum.Provider `json:"type"`
+}
+
+func NewDbUpdateProvider(id string, typ spenum.Provider) *DbUpdateProvider {
+	return &DbUpdateProvider{
+		DbUpdates: DbUpdates{
+			Id:      id,
+			Updates: make(map[string]interface{}),
+		},
+		Type: typ,
 	}
+
 }
 
-type StakePoolId struct {
+type Provider struct {
```

Comment:
In what way? `dbs.Provider` in a different package from the other Provider objects, so as far as golang code is concerned. It is not stored directly in a database but instead used to transfer messages between smart contracts and the event database. Plus the old name was just wrong.

Reply:
Just logically, we have too much Providers in different packages

Reply:
Not quite sure what you mean, but I renamed Provider to ProviderID. 

The field names were chosen to match the json tags, ProviderId for provider_id. Although it might make sense to change ProviderId to ID as the struct name implies the provider part. Anyway changed to 
```go
type ProviderID struct {
	ID   string          `json:"provider_id"`
	Type spenum.Provider `json:"provider_type"`
}
```